src/translate/pddl_parser/parsing_functions.py

```diff
@@ -188,121 +314,229 @@ def add_effect(tmp_effect, result):
-    action_tag = next(iterator)
-    assert action_tag == ":action"
-    name = next(iterator)
-    parameters_tag_opt = next(iterator)
-    if parameters_tag_opt == ":parameters":
-        parameters = parse_typed_list(next(iterator),
-                                      only_variables=True)
-        precondition_tag_opt = next(iterator)
-    else:
-        parameters = []
-        precondition_tag_opt = parameters_tag_opt
-    if precondition_tag_opt == ":precondition":
-        precondition_list = next(iterator)
-        if not precondition_list:
-            # Note that :precondition () is allowed in PDDL.
-            precondition = pddl.Conjunction([])
+        return pddl.SimpleEffect(parse_literal(context, alist, {}, predicate_dict))
+
+
+def parse_expression(context, exp):
+    syntax = "'POSITIVE_NUMBER' or '(FUNCTION VARIABLES*)'"  # TODO: final case?
+    with context.layer("Parsing expression"):
+        if isinstance(exp, list):
+            if len(exp) < 1:
+                context.error(f"Expecting {syntax}.")
+            functionsymbol = exp[0]
+            return pddl.PrimitiveNumericExpression(functionsymbol, exp[1:])
+        elif exp.replace(".", "").isdigit():
+            return pddl.NumericConstant(float(exp))
+        elif exp[0] == "-":
+            context.error("Negative numbers are not supported")
         else:
-            precondition = parse_condition(
-                precondition_list, type_dict, predicate_dict)
-        effect_tag = next(iterator)
-    else:
-        precondition = pddl.Conjunction([])
-        effect_tag = precondition_tag_opt
-    assert effect_tag == ":effect"
-    effect_list = next(iterator)
-    eff = []
-    if effect_list:
+            return pddl.PrimitiveNumericExpression(exp, [])
+
+
+def parse_assignment(context, alist):
+    syntax = "'({=,increase} EXPRESSION EXPRESSION)'"
+    with context.layer("Parsing Assignment"):
+        if len(alist) != 3:
+            context.error(f"Expecting {syntax}.")
+        op = alist[0]
+        head = parse_expression(context, alist[1])
+        exp = parse_expression(context, alist[2])
+        if op == "=":
+            return pddl.Assign(head, exp)
+        elif op == "increase":
+            return pddl.Increase(head, exp)
+        else:
+            context.error("Assignment operator not supported. Use '=' or 'increase'.")
+
+
+def parse_action(context, no, alist, type_dict, predicate_dict):
+    syntax = "(:action NAME " \
+             "[:parameters PARAMETERS]? " \
+             "[:precondition PRECONDITION]? " \
+             ":effect EFFECT)"
+    with context.layer(f"Parsing {no}. action"):
+        context.check_list(alist, min_size=4, finalmsg=f"Expecting '{syntax}'.")
+        iterator = iter(alist)
+        action_tag = next(iterator)
+        assert action_tag == ":action"
+        name = next(iterator)
+        context.check_string(name, item_name="Action name")
+    with context.layer(f"Parsing {no}. action '{name}'"):
```

Comment:
I created a new layer (and closed the old one), because at this place in code,  we have additional information which is useful for the user namely the action name. I would keep this layer due to the added value of the action name for debugging.

Reply:
Then maybe rename the first layer to "Parsing {no}. action name". (I'd also get rid of `no` here but let's discuss that in the other thread the reasoning here is the same.)